### ISSUE TITLE
Support webP transform with the image proxy transform API

### ISSUE THREAD

**Issue by dohomi** [nhost/nhost#2508]
**Is your feature request related to a problem? Please describe.**
Returning webP for JPG, PNG and other file format would significantly reduce the file size and accelerate the CDN API by lowering the transferred data size.

**Describe the solution you'd like**
It would be great if images be optimised when `q=70` or a different query param is been used.

**Additional context**
WebP is supported by the 96% of browsers based on
https://caniuse.com/webp

**Comment by dbarrosop** [comment]
Automatic conversion between different formats is something we probably don't want to do but we could add a `?f=webp` or similar to request the output format.

**Comment by jovermier** [comment]
I have been investigating this exact idea @dohomi over the past week. Thank you for creating this post. I would also like the ability to request a specific file type from nhost. It looks like nhost uses sharp package which already supports this feature. The search param `f=webp` would be a great option.

I was also considering doing the conversion to webp when the image is uploaded so storage only has webp images in it thus cutting down on the storage usage and request image transformation execute time.

**Comment by dbarrosop** [comment]
We don't use sharp (wrong programming language :P) but it doesn't matter, the library we use is also built on top of libvips and has support for this. If someone wants to contribute this feature that'd be great, happy to help guiding. Otherwise we will get to this when we can.

**Comment by dohomi** [comment]
I guess the best would be something like a `?o=1` (optimise=true) and based on the supported request header the image proxy would return WebP, AVIF or any other image format supported by the browser.

**Comment by dbarrosop** [comment]
Optimize would mean different things for different people so I think being explicit is the best path forward. Also, the backend has no access to browser information so it'd be impossible to know what's supported and what's not.

**Comment by jovermier** [comment]
To clarify @dohomi, are you suggesting that if `?o=1` is present then the server should automatically convert the image to the most ideal output based on the request headers?

I grabbed this request header example from my browser:
```
Accept: image/avif,image/webp,image/apng,image/svg+xml,image/*,*/*;q=0.8
```

**Comment by dohomi** [comment]
@jovermier yes I think any path, either `o`=>optimise or `c`=>compress would convert the image in following order:
AVIF, WebP, no manipulation based on `Accept` header.

I would assume that the image proxy should have access to the browser `Accept` headers but @dbarrosop knows it probably better.

**Comment by dohomi** [comment]
@jovermier any chance that this request will get attention?

**Comment by dbarrosop** [comment]
While I think this might be a fine addition it isn't currently scheduled due to other priorities. If someone wants to help out and contribute the feature I'd be more than glad to help out.

**Comment by dohomi** [comment]
@dbarrosop where is the image transform package located?

**Comment by dohomi** [comment]
@dbarrosop I updated my project to the 0.7.0 beta and the format=auto works flawlessly! I tested it over a few of my image assets and the reduction was around 25-30% reduction! That is a huge win.

It would be great if the SDK could get updated to allow the `format` option, currently it converts it converts `format=auto` into `f=auto` which doesn't trigger the conversion.

**Comment by dohomi** [comment]
What also realised is: when quality is set to lets say 85, the combination of quality+format results into a bigger asset size than just format. For my images it seems the best reduction is using at the moment format=webp without a quality conversion.

**Comment by dbarrosop** [comment]
> But the format=auto does not seem to work, I haven't seen a change to webp or avif.

can you check which headers are being sent with the request? if there is no Accept headers we default to not transforming the image at all. 

> It would be great if the SDK could get updated to allow the format option, currently it converts format=* into f=* which doesn't trigger the conversion.

Yes, the SDK needs some work and a PR would be more than welcome 😉 We need to update the types to include format, add the docstring and maybe patch the issue you are mentioning. Either that or rename `format` with just `f` for consistency...

> What I also realised: when quality is set to lets say 85, the combination of quality+format results into a bigger asset size than just format.

The issue with quality is that they have different meanings in different formats so I am not surprised if you see inconsistencies between formats when it comes to size.

**Comment by dohomi** [comment]
I checked the headers, the `Accept` header allows avif but still the response remains png

<img width="594" alt="Image" src="https://github.com/user-attachments/assets/cebdb401-0668-4b64-b1aa-fe263bd6452f" />

**Comment by dohomi** [comment]
This is the output if I append `format=avif` which reduces from ~36000 => ~8100 which is just awesome..

<img width="593" alt="Image" src="https://github.com/user-attachments/assets/4fb877c0-503e-462c-a69c-bb1f11262209" />

**Comment by dbarrosop** [comment]
Ok, two things:
1. Renamed `format` to `f`
2. `Accept` header should work now

Could you test again, please? Thanks

**Comment by dohomi** [comment]
Yes, both changes work now as expected. `f=auto` takes accept into account.
